File: inspect_db.js
import Database from 'better-sqlite3';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const dbPath = path.join(__dirname, 'data', 'pipeline.db');
const db = new Database(dbPath);

console.log('=== CHANNELS ===');
console.log(db.prepare('SELECT * FROM channels').all());

console.log('\n=== VIDEOS ===');
const videos = db.prepare('SELECT id, channel_id, original_filename, created_at FROM videos').all();
console.log(videos);

console.log('\n=== SCHEDULED POSTS ===');
const posts = db.prepare('SELECT id, channel_id, video_id, youtube_video_id, status, scheduled_at, created_at FROM scheduled_posts').all();
console.log(posts);

db.close();

File: scratch/inspect_db.py
import sqlite3
import json

db_path = "/var/www/youtube-pipeline/data/pipeline.db"
conn = sqlite3.connect(db_path)
conn.row_factory = sqlite3.Row
cursor = conn.cursor()

# Get all scheduled posts for channel 4 (Senpai)
cursor.execute("""
    SELECT sp.id, sp.title, sp.scheduled_at, sp.status, sp.youtube_video_id, sp.is_premiere
    FROM scheduled_posts sp
    WHERE sp.channel_id = 4
    ORDER BY sp.scheduled_at ASC
""")
rows = [dict(r) for r in cursor.fetchall()]

print("--- SENPAI SCHEDULED POSTS ---")
print(json.dumps(rows, indent=2))
